persistence-api/src/main/java/io/stargate/db/Persistence.java

```diff
@@ -15,96 +15,162 @@
  */
 package io.stargate.db;
 
+import com.datastax.oss.driver.shaded.guava.common.util.concurrent.Uninterruptibles;
```

Comment:
Do we want to expose driver class in persistence API? I believe it would be preferable to hide them (even if we use the internally). For SQL, for example, C* drivers may not be relevant at all.

Reply:
Well, on the flip side we really don't want to use guava at this layer. Is there an alternative to `Uninterruptibles` or maybe we should shade our own guava?

Reply:
I see that it's used a lot in persistence API, but comes from a dedicated artifact (`java-driver-shaded-guava`), so it's probably fine.

Reply:
That artifact is intended for internal use in the driver only, so I would not rely on it. It's not guaranteed to stay there forever (at some point there was some talk of forking the required classes into the driver codebase directly and removing it).

Reply:
So there is 2 sub-parts here:

The class you're pointing to is indeed a guava class, imported from the shaded version inside the driver. I agree using that is quite dodgy, but again, this is actually copied code from `DataStore` and I simply didn't change it. Imo, if we want to use Guava, we should add guava as a direct dependency and use it (but the other option could be to avoid it altogether, I don't think it's that used in `persistence-api` so far; don't know if we care about minimizing dependencies). Anyway, I suggested in another comment to delay cleaning up `waitForSchemaAgreement` to a followup and that's what use it, so maybe leave that to that followup? Or if that bother you guys, happy to add guava as dependency and replace.

The other point though is that we _do_ expose driver classes in the persistence API. And it is not new to this patch either. For instance, the `Column.ColumnType` classes exposes `TypeCodec` from the driver. Admittedly, this patch does expose driver classes a bit more because I've reused `GettableByIndex`/`GettableByName` in `Row` to allow reusing quite a bit of boring code. But to answer the more general question, I do think we should absolutely reuse the abstraction that already exists in the drivers, are public, and have already had the test of time. In fact, I suggest we use more of the driver code: the `DataStore` is essentially a driver-like interface to persistence, so it uses a lot of the same concepts (Row, ResultSet, Columns, ...) and I think it's a shame we've rewritten quite a few of them. In fact, I plan to look soonish if we can consider reusing much more of the driver interfaces for things like `ResulSet` and all the schema classes.

Reply:
Actually, re-shaded-guava: I just realized how often we were accessing it in the `persistence-api` module, and we use it in almost every other module anyway, so I'm going to add the dependency and replace the imports now.

Reply:
Well well. Turns out there is a reason the `persistence-api` was relying on the shaded jar. C* uses a antiquated version of guava (18.0) and having a newer version in the classpath breaks it (and if I instead "force" the guava version of C* for the `persistence-api`, this doesn't work either because it doesn't have some of the method we use, like `Preconditions#checkArgument`, which is a shame).

Tl;dr, I'll revert to the shaded guava for now. As @dimas-b mentions, we explicitly import it, so it's not "that" terrible. We can find a better option later if we're bothered.